modules/021-cni-cilium/hooks/discovery_cni_exclusive.go

```diff
@@ -44,16 +41,31 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
 			},
 			FilterFunc: daemonsetFilter,
 		},
+		{
+			Name:       "sdn-cni-daemonset",
+			ApiVersion: "apps/v1",
+			Kind:       "DaemonSet",
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"sdn-agent"},
+			},
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"d8-sdn"},
+				},
+			},
+			FilterFunc: daemonsetFilter,
+		},
 	},
-}, discoveryIsIstioCNIEnabled)
+}, discoveryIsExclusiveCNIPluginEnabled)
 
 func daemonsetFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
 	return obj.GetName(), nil
 }
 
-func discoveryIsIstioCNIEnabled(input *go_hook.HookInput) error {
+func discoveryIsExclusiveCNIPluginEnabled(input *go_hook.HookInput) error {
 	istioCniDaemonSets := input.NewSnapshots.Get("istio-cni-daemonset")
-	isIstioCNIEnabled := len(istioCniDaemonSets) != 0
-	input.Values.Set("cniCilium.internal.isIstioCNIEnabled", isIstioCNIEnabled)
+	sdnCniDaemonSets := input.NewSnapshots.Get("sdn-cni-daemonset")
+	isEnabled := len(istioCniDaemonSets) != 0 || len(sdnCniDaemonSets) != 0
+	input.Values.Set("cniCilium.internal.exclusiveCNIPlugin", isEnabled)
```

Comment:
Here the `cniCilium.internal.exclusiveCNIPlugin` means 'is something enabled'.
The logic must be:
```
if len(istioCniDaemonSets) != 0 || len(sdnCniDaemonSets) != 0 {
  	input.Values.Set("cniCilium.internal.exclusiveCNIPlugin", false)
} else {
    eCNIP := input.Values.Get("cniCilium.exclusiveCNIPlugin")
    input.Values.Set("cniCilium.internal.exclusiveCNIPlugin", eCNIP)
}
```